presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -989,7 +989,13 @@ public Node visitSpecialDateTimeFunction(SqlBaseParser.SpecialDateTimeFunctionCo
     @Override
     public Node visitExtract(SqlBaseParser.ExtractContext context)
     {
-        return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), Extract.Field.valueOf(context.identifier().getText().toUpperCase()));
+        String field = context.identifier().getText();
+        try {
+            return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), Extract.Field.valueOf(field.toUpperCase()));
+        }
+        catch (IllegalArgumentException e) {
+            throw new ParsingException("Invalid extract field: " + field, null, context.getStart().getLine(), context.getStart().getCharPositionInLine());
```

Comment:
Change the message to something like "Invalid field for EXTRACT expression", or "Expected EXTRACT field to be one of xxx, yyy, zzz, ..."